From 85a6e4977cf307dff5ebefa93bb08bc10c9cfb16 Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Wed, 11 May 2016 23:25:18 +0200
Subject: wininet: Return synchronously from InternetReadFile if there is no
 more data to read.

Similar to 87fffa1ebfb71fe7a34aa989fa88f85409dbb62f.

Signed-off-by: Sebastian Lackner <sebastian@fds-team.de>
---
 dlls/wininet/http.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/wininet/http.c b/dlls/wininet/http.c
index 646619b..c7d857e 100644
--- a/dlls/wininet/http.c
+++ b/dlls/wininet/http.c
@@ -3242,7 +3242,7 @@ static DWORD HTTPREQ_ReadFile(object_header_t *hdr, void *buffer, DWORD size, DW
 
         if (TryEnterCriticalSection( &req->read_section ))
         {
-            if (get_avail_data(req))
+            if (get_avail_data(req) || end_of_read_data(req))
             {
                 res = HTTPREQ_Read(req, buffer, size, read);
                 LeaveCriticalSection( &req->read_section );
-- 
2.8.0

